test/concourse/model/traits/filter_test.py

```diff
@@ -0,0 +1,140 @@
+import dataclasses
+import typing
+
+import pytest
+
+import concourse.model.traits.filter as examinee
+
+
+# Dummy dataclass for testing
+@dataclasses.dataclass
+class Dummy:
+    name: str
+
+
+@pytest.fixture
+def dummy():
+
+    def _dummy(
+        name: str = 'Dummy',
+    ):
+        return Dummy(
+            name=name,
+        )
+
+    return _dummy
+
+
+class TestMatchingFilter:
+    @pytest.fixture
+    def rule(self):
+        def _rule(
+            target: str = '',
+            expression: str = '',
+            matching_semantics: str = 'include',
+        ):
+            return examinee.MatchingConfigRule(
+                target=target,
+                expression=expression,
+                matching_semantics=examinee.ComponentFilterSemantics(
+                    matching_semantics
+                ),
+            )
+
+        return _rule
+
+    @pytest.fixture
+    def matching_config(self):
+        def _matching_config(
+            name: str = 'TestConfig',
+            rules: typing.List[examinee.MatchingConfigRule] = [],
+        ):
+            return examinee.MatchingConfig(
+                name=name,
+                rules=rules,
+            )
+
+        return _matching_config
+
+    def test_unspecific_target_fails(self, dummy, matching_config, rule):
+        config = matching_config(
+            rules=[
+                rule(
+                    target='name',
+                    expression='TestComponent',
+                    matching_semantics='include',
+                ),
+            ],
+        )
+        filter = examinee.filter_for_matching_config(config)
+        component_dummy = dummy(name='TestComponent')
+        with pytest.raises(ValueError):
+            filter(component_dummy, None)
+
+    def test_component_attr_included(self, dummy, matching_config, rule):
+        config = matching_config(
+            rules=[
+                rule(
+                    target='component.name',
+                    expression='TestComponent',
+                    matching_semantics='include',
+                ),
+            ],
+        )
+        filter = examinee.filter_for_matching_config(config)
+        component_dummy = dummy(name='TestComponent')
+        assert filter(component_dummy, None) == True
```

Comment:
nit: I think it is a bad practise to "shadow" builtin names (filter, map, ..)
next nit: boolean values should not be used in comparisons, see https://peps.python.org/pep-0008/
-> `assert filter_for_matching_config(cfg)`